internal/fs/inode/file.go

```diff
@@ -119,10 +119,13 @@ func NewFileInode(
 		attrs:          attrs,
 		localFileCache: localFileCache,
 		contentCache:   contentCache,
-		src:            storageutil.ConvertObjToMinObject(o),
 		local:          localFile,
 		unlinked:       false,
 	}
+	minObjPtr := storageutil.ConvertObjToMinObject(o)
+	if minObjPtr != nil {
+		f.src = *minObjPtr
```

Comment:
given the old code always expects src to be non empty, so we should follow that and panic if minObjPtr is nil, otherwise this can have unexpected results later if someone makes changes.

Reply:
As discussed offline, when nothing will be assigned to f.src, it will have empty gcs.MinObject (zero value of struct). This is in sync with original behavior of ConvertObjToMinObject as it returned empty gcs.MinObject when o is nil.
But we will still make this change for readability.